src/server.ts

```diff
@@ -11,8 +11,9 @@ import { EBB } from "./ebb";
 import { Device, PenMotion, Motion, Plan } from "./planning";
 import { formatDuration } from "./util";
 import { autoDetect } from '@serialport/bindings-cpp';
+import * as self from './server'  // for mocking
```

Comment:
Is there a link to documentation for this? Its a new pattern to me, but otherwise this looks good!

Reply:
Exporting the function was the only way I found that allowed `jest` to [mock it](https://jestjs.io/docs/mock-functions).  It doesn't appear to recurse (presumably after the first `jest` "grabs" it and stops the recursion?)

I have no idea if it's a good pattern or not; it was trial-and-error.